# Introduction to the Bayes Factor and Posterior Odds:
# The Completion of Ideas

In [1]:
import pandas as pd
import numpy as np
from matplotlib import pyplot as plt

In [2]:
from scipy.stats import norm, beta, uniform

# Exercise 1

<b>
In the dice problem, assume your friend made a mistakeand suddenly realized that there were, in fact, two loaded dice and only one fair die. How does this change the prior, and therefore the posterior odds, for our problem?
    
Are you more willing to believe that the die being rolled is the loaded die?
</b>

We have a change in the prior odds. Now:

P(H<sub>1</sub>) = 2/3

P(H<sub>2</sub>) = 1/3


Our pior odds becomes:

O(H<sub>1</sub>) = P(H<sub>1</sub>) / P(H<sub>2</sub>) = <sup>2/3</sup> / <sub>1/3</sub> = 2

posterior odds = O(H<sub>1</sub>) * <sup>P(D|H<sub>1</sub>)</sup> / <sub>P(D|H<sub>2</sub>)</sub>

In [2]:
2 * 3.77

7.54

I am more willing to believe the die being rolled is a loded die. However the posterior odds are not very high, more evidence will help to make a better decision

# Exercise 2

Suppose you go to the doctor, and after having your ears cleaned you notice that your symptoms of hearing loss and tinnitus persist. Even worse, you have a new symptom: vertigo. The doctor proposes another possible explanation, **labyrinthitis**, which is a viral infection of the inner ear in which 98 percent of cases involve vertigo.

Howerver, hearing loss and tinnitus are less common in this disease; hearing loss occurs only 30 percent of the time, and tinnitus occurs only 28 percent of the time. Vertigo is also a possible symptom of vestibular schwanoma, but occurs in only 49 percent of cases.

In the general population, 35 people per million contract labyrinthitis annually.

What is the posterior odds when you compare the hypothesis that you have labyrinthitis against the hypothesis that you have vestibular schwannoma?

H<sub>1</sub> = labyrinthitis

P(H<sub>1 Hearing Loss</sub>) = 0.3

P(H<sub>1 Teinnitus</sub>)  = 0.28

P(H<sub>1 Vertigo</sub>) = 0.98

P(H<sub>1 Vertigo</sub>) = 0.98

P(D | H<sub>2</sub>) = 0.94 * 0.83 * 0.49

---

H<sub>2</sub> = vestibular schwanoma

P(H<sub>2 Hearing Loss</sub>) = 0.94

P(H<sub>2 Teinnitus</sub>)  = 0.83

P(H<sub>2 Vertigo</sub>) = 0.49

P(D | H<sub>1</sub>) = 0.3 * 0.28 * 0.98

---

Bayes Factor = <sup>P(D | H<sub>1</sub>)</sup> / <sub>P(D | H<sub>2</sub>)</sub>

---

In [15]:
bayes_factor = (0.3 * 0.28 * 0.98) / (0.94 * 0.83 * 0.49)

In [16]:
bayes_factor

0.21532940271725204


Prior Odds = O(H<sub>1</sub>) = <sup>P(H<sub>1</sub>)</sup> / <sub>P(H<sub>2</sub>)</sub>

P(H<sub>1</sub>) = 35 / 1,000,000

P(H<sub>2</sub>) = 11 / 1,000,000

In [17]:
prior_odds = (35 / 1_000_000) / (11 / 1_000_000)

In [18]:
prior_odds

3.1818181818181817

Posterior Odds = Prior Odds * Bayes Factor

In [19]:
odds_of_labyrinthitis = prior_odds * bayes_factor

In [20]:
odds_of_labyrinthitis

0.6851390086458019

In [13]:
odds_of_vestibular_schwannoma = 1 / odds_of_labyrinthitis

In [21]:
odds_of_vestibular_schwannoma

1.4595578231292514

Comparing hypothesis of having labyrinthitis against vestibular schwannoma we can see that we are 1.46 times more likely to have vestibular schwannoma